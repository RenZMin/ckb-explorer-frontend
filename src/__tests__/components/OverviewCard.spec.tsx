import { ReactElement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import OverviewCard, { OverviewItem } from '../../components/Card/OverviewCard'

describe('OverviewCard Component', () => {
  let component: ReactElement

  beforeAll(() => {
    const items = [
      {
        title: 'title1',
        content: 'content1',
      },
      {
        title: 'title2',
        content: <div>content2</div>,
      },
      {
        title: 'title3',
        content: 'content3',
      },
      {
        title: 'title4',
        content: <div>content4</div>,
      },
    ]
    component = <OverviewCard items={items} />
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find('.overview_content__left_items')).toHaveLength(1)
    expect(wrapper.find('.overview_content__right_items')).toHaveLength(1)
    expect(wrapper.find(OverviewItem)).toHaveLength(4)
  })
})
