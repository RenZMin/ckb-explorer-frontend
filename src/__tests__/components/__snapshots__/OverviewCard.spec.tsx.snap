// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewCard Component shallow renders 1`] = `
<div
  className="sc-AxjAm dfNVFO"
>
  <div
    className="overview__separate"
  />
  <div
    className="sc-AxirZ hXUsTm"
  >
    <div
      className="overview_content__left_items"
    >
      <div
        className="sc-AxiKw eGZoJv"
      >
        <div
          className="overview_item__title__panel"
        >
          <div
            className="overview_item__title"
          >
            <span>
              title1
            </span>
          </div>
        </div>
        <div
          className="overview_item__value"
        >
          <span>
            content1
          </span>
        </div>
      </div>
      <div
        className="sc-AxiKw dzAEis"
      >
        <div
          className="overview_item__title__panel"
        >
          <div
            className="overview_item__title"
          >
            <span>
              title3
            </span>
          </div>
        </div>
        <div
          className="overview_item__value"
        >
          <span>
            content3
          </span>
        </div>
      </div>
    </div>
    <div
      className="overview_content__right_items"
    >
      <div
        className="sc-AxiKw eGZoJv"
      >
        <div
          className="overview_item__title__panel"
        >
          <div
            className="overview_item__title"
          >
            <span>
              title2
            </span>
          </div>
        </div>
        <div
          className="overview_item__value"
        >
          <span>
            <div>
              content2
            </div>
          </span>
        </div>
      </div>
      <div
        className="sc-AxiKw dzAEis"
      >
        <div
          className="overview_item__title__panel"
        >
          <div
            className="overview_item__title"
          >
            <span>
              title4
            </span>
          </div>
        </div>
        <div
          className="overview_item__value"
        >
          <span>
            <div>
              content4
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
